src/map/pc.c

```diff
@@ -9206,10 +9206,6 @@ static int pc_jobchange(struct map_session_data *sd, int class, int upper)
 	if (sd->vd.body_style)
 		clif->changelook(&sd->bl,LOOK_BODY2,sd->vd.body_style);
 
-	//Update skill tree.
-	pc->calc_skilltree(sd);
-	clif->skillinfoblock(sd);
```

Comment:
These two lines are called after `status_calc_pc` in line 9252, and removing them also fixes the same scenario when jobchanging.